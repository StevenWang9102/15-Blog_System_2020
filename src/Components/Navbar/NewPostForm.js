import React, {useState} from "react";

export const NewPostForm = props => {
  const [alertText, setAlertText] = useState("");

  return (
    <form>
      <fieldset>
        {/* ------- Article Title ------- */}
        <fieldset className='form-group'>
          <input
            type='text'
            className='form-control form-control-lg'
            onChange={event => props.setTitle(event.target.value)}
            value={props.title}
            placeholder='Article Title'></input>
        </fieldset>

        {/* ------- Article Description ------- */}
        <fieldset className='form-group'>
          <input
            type='text'
            className='form-control'
            value={props.description}
            onChange={event => props.setDescription(event.target.value)}
            placeholder="What's this article about?"></input>
        </fieldset>

        {/* ------- Article Content ------- */}
        <fieldset className='form-group'>
          <textarea
            className='form-control'
            rows='8'
            value={props.content}
            onChange={event => props.setContent(event.target.value)}
            placeholder='Write your article (in markdown)'></textarea>
        </fieldset>

        {/* ------- Article Tags ------ */}
        <fieldset className='form-group'>
          <input
            type='text'
            className='form-control'
            value={props.tags}
            onChange={event => props.setTags(event.target.value)}
            placeholder='Enter tags'></input>
          <div className='tag-list'></div>
        </fieldset>

        <button
          className='btn btn-lg pull-xs-right btn-primary'
          type='button'
          onClick={() => {
            if(props.title.length===0||props.description.length===0||props.content.length===0||props.tags.length===0){
              setAlertText("* every input can not be blank")
            }else{
              props.setLoading("LOADING");
              props.onPostArticleClicked(
              props.title,
              props.description,
              props.content,
              props.tags,
              props.currentSlug
            );}
          }}>
          Publish Article
        </button>
        <div className='sign_Alert'> {alertText} </div>
      </fieldset>
    </form>
  );
};
