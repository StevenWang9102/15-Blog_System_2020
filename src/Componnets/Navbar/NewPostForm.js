import React from "react";

export const NewPostForm = props => {
  return (
    <form>
      <fieldset>
        {/* ------- Article Title ------- */}
        <fieldset className='form-group'>
          <input
            type='text'
            className='form-control form-control-lg'
            onChange={event => props.setTitle(event.target.value)}
            value={props.title}
            placeholder='Article Title'></input>
        </fieldset>

        {/* ------- Article Description ------- */}
        <fieldset className='form-group'>
          <input
            type='text'
            className='form-control'
            value={props.description}
            onChange={event => props.setDescription(event.target.value)}
            placeholder="What's this article about?"></input>
        </fieldset>

        {/* ------- Article Content ------- */}
        <fieldset className='form-group'>
          <textarea
            className='form-control'
            rows='8'
            value={props.content}
            onChange={event => props.setContent(event.target.value)}
            placeholder='Write your article (in markdown)'></textarea>
        </fieldset>

        {/* ------- Article Tags ------ */}
        <fieldset className='form-group'>
          <input
            type='text'
            className='form-control'
            value={props.tags}
            onChange={event => props.setTags(event.target.value)}
            placeholder='Enter tags'></input>
          <div className='tag-list'></div>
        </fieldset>

        <button
          className='btn btn-lg pull-xs-right btn-primary'
          type='button'
          onClick={() => {
            props.setLoading("LOADING");
            props.onPostArticleClicked(
              props.title,
              props.description,
              props.content,
              props.tags,
              props.currentSlug
            );
          }}>
          Publish Article
        </button>
      </fieldset>
    </form>
  );
};
