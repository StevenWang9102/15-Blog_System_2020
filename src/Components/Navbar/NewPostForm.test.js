import React from "react";
import { NewPostForm } from "./NewPostForm";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <NewPostForm
        title='test'
        description='test'
        content='test'
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
