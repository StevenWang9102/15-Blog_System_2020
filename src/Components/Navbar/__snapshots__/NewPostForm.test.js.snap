// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form>
  <fieldset>
    <fieldset
      className="form-group"
    >
      <input
        className="form-control form-control-lg"
        onChange={[Function]}
        placeholder="Article Title"
        type="text"
        value="test"
      />
    </fieldset>
    <fieldset
      className="form-group"
    >
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="What's this article about?"
        type="text"
        value="test"
      />
    </fieldset>
    <fieldset
      className="form-group"
    >
      <textarea
        className="form-control"
        onChange={[Function]}
        placeholder="Write your article (in markdown)"
        rows="8"
        value="test"
      />
    </fieldset>
    <fieldset
      className="form-group"
    >
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Enter tags"
        type="text"
      />
      <div
        className="tag-list"
      />
    </fieldset>
    <button
      className="btn btn-lg pull-xs-right btn-primary"
      onClick={[Function]}
      type="button"
    >
      Publish Article
    </button>
    <div
      className="sign_Alert"
    >
       
      
       
    </div>
  </fieldset>
</form>
`;
